proposals/0016-simd-application-fees.md

```diff
@@ -0,0 +1,438 @@
+---
+simd: '0016'
+title: Application Fees
```

Comment:
This proposal is really talking about "Account write-access fees". I'd rather the title and overall discussion be framed in that direction.

Reply:
Naming the feature account write fees has some negative implications over the authority of these fees. Application fees because fees are configured and set by application developers and will go to accounts controlled by application developers.

Reply:
I tend to agree with this in light of the fact we may at some point be introducing a base fee multiplier based on the number of writable accounts in a TX. That multiplier is more of an "account write access fee multiplier," would be confusing to have both concepts. This is intended for dapp developers to set so it makes more sense to name them application fess, especially if it gives the dapp the ability to charge for "incorrect behavior" rather than simply just access to the writable account. That ability to anlyze and charge for behavioral access patterns makes it sound much more like an "application" fee than a simple "write-access" fee.

Reply:
there is a proposal for dynamic account write frees which is not specific to the program, but chain-wide, so "program" is not superfluos. i would prefer a very distinctive term to easily communicate this concept to dapp developers. program account write fees is just very close to dynamic account write fees.

Reply:
changed to Program -rebateble account write fees.